src/scripts/data_processing/generate_data/generate.py

```diff
@@ -84,6 +91,140 @@ def get_arg_parser(cls, parser=None):
+            fmt="%1.3f",
+        )
+
+
+    def generate_regression_tasks(self, args):
+        """Create generation tasks based on arguments"""
+        train_partition_size = args.train_samples // args.train_partitions
+        test_partition_size = args.test_samples // args.test_partitions
+        inferencing_partition_size = args.inferencing_samples // args.inferencing_partitions
+
+        batch_size = min(
+            10000,
+            train_partition_size,
+            test_partition_size,
+            inferencing_partition_size,
+        )
+
+        self.logger.info(f"Using batch size {batch_size}")
+
+        generator = RegressionDataGenerator(
+            batch_size=batch_size,
+            n_features=args.n_features,
+            n_informative=args.n_informative,
+            n_targets=1,
+            bias=0.0,
+            noise=0.0,
+            seed=args.random_state,
+        )
+
+        # add train partitions to list of tasks
+        for i in range(args.train_partitions):
+            self.generation_tasks.append(
+                (os.path.join(args.output_train, f"train_{i}.txt"), generator, train_partition_size//batch_size)
+            )
+
+        # add test partitions to list of tasks
+        for i in range(args.test_partitions):
+            self.generation_tasks.append(
+                (os.path.join(args.output_test, f"test_{i}.txt"), generator, test_partition_size//batch_size)
+            )
+
+        # add inferencing partitions to list of tasks
+        for i in range(args.inferencing_partitions):
+            self.generation_tasks.append(
+                (os.path.join(args.output_inference, f"inference_{i}.txt"), generator, inferencing_partition_size//batch_size)
+            )
+
+    def execute_tasks(self):
+        # show some outputs first
+        for output_file_path, generator, batches in self.generation_tasks:
+            self.logger.info(f"Will generate output {output_file_path} with {batches} batches")
+
+        # generate each data outputs
+        for output_file_path, generator, batches in self.generation_tasks:
+            self.logger.info(f"Opening file {output_file_path} for writing...")
+
+            # create/erase file
+            with open(output_file_path, "w") as output_file:
+                output_file.write("")
+
+            # iterate and append
+            for i in range(batches):
+                with self.metrics_logger.log_time_block("time_data_generation_batch"):
+                    X,y = generator.generate()
+                    y = numpy.reshape(y, (y.shape[0], 1))
+                    data = numpy.hstack((y, X))  # keep target as column 0
+
+                with self.metrics_logger.log_time_block("time_data_saving_batch"):
+                    with open(output_file_path, "a") as output_file:
+                        numpy.savetxt(
+                            output_file,
+                            data,
+                            delimiter=",",
```

Comment:
It might be a good time to give the option to use tsv files. This way we have tested the benchamrk code against the most common data format in WebXT

Reply:
good idea, I implemented a new delimiter parameter.